source/common/http/http1/codec_impl.cc

```diff
@@ -878,6 +879,30 @@ int ServerConnectionImpl::onHeadersComplete() {
           "http/1.1 protocol error: request headers failed spec compliance checks");
     }
 
+    // https://tools.ietf.org/html/rfc7230#section-3.3.3
+    // If a message is received with both a Transfer-Encoding and a
+    // Content-Length header field, the Transfer-Encoding overrides the
+    // Content-Length. Such a message might indicate an attempt to
+    // perform request smuggling (Section 9.5) or response splitting
+    // (Section 9.4) and ought to be handled as an error. A sender MUST
+    // remove the received Content-Length field prior to forwarding such
+    // a message downstream.
+
+    // Reject request with Http::Code::BadRequest by default or remove Content-Length header
+    // and serve request if allowed by http1 codec settings.
+    if (parser_.uses_transfer_encoding != 0 &&
```

Comment:
Sorry why this uses_transfer_encoding check? Couldn't you just do `(parser_.flags & F_CHUNKED)`? Or you are checking for any transfer encoding? Can you add more comments?

Reply:
I was following RFC here (any TE, not only `chunked`):
```
If a message is received with both a Transfer-Encoding and a
Content-Length header field, the Transfer-Encoding overrides the
Content-Length.
```